File: hackerRank/firstDuplicate/firstDuplicate.js
function firstDuplicate(a) {
    let lowIndex = [],
        index,
        currArr,
        i;

    for (i = 0; i < a.length - 1; i++) {
        if (i >= lowIndex[0]) {
            break;
        }
        currArr = a.slice(i + 1)
        if (currArr.indexOf(a[i]) > -1) {
            index = currArr.indexOf(a[i]) + i + 1;
            if (index < lowIndex[0] || lowIndex.length == 0) {
                lowIndex.pop();
                lowIndex.push(index)
                // a.splice(0, index);
            }
        }
    }
    return lowIndex.length ? a[lowIndex[0]] : -1;
}

File: hackerRank/firstDuplicate/firstDuplicate.rb
def firstDuplicate(a)
    let lowIndex = [],
        i,
        j;

    for (i = 0; i < a.length; i++) {
        if (lowIndex[0] < arr[i]) {
            return lowIndex[0];
        }
        for (j = i + 1; j < a.length; j++) {
            if (arr[i] == arr[j]) {
                lowIndex.pop()
                lowIndex.push(j);
                break;
            }
        }
    }
    return -1;
end
